moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -1184,40 +1184,40 @@ class MoveGroup::MoveGroupImpl
 }
 }
 
-moveit::planning_interface::MoveGroup::MoveGroup(const std::string &group_name, const boost::shared_ptr<tf::Transformer> &tf, const ros::WallDuration &wait_for_servers)
+moveit::planning_interface::MoveGroupInterface::MoveGroupInterface(const std::string &group_name, const boost::shared_ptr<tf::Transformer> &tf, const ros::WallDuration &wait_for_servers)
```

Comment:
> From the syntax examples on cppreference I am advocating #1 and you are advocating we stick with #4, correct/

Yes

> > you see that the object file provides the symbol foo::func

> is that bad?

:laughing:
No it's not bad. It's the case; whichever style is used.
I was just pointing out that, when using #1 from your reference, the symbol names in the actual code are not the ones that are exported in the compiled libraries (in the compiled libraries the symbols are *always* prefixed by their namespaces).
I (and many others) prefer to have them the same in code and in the libs for ease of debugging. Thus I consider it best-practice.